### DISCUSSION TITLE
http://www.andrewheiss.com/blog/2017/09/27/working-with-r-cairo-graphics-custom-fonts-and-ggplot/

### DISCUSSION THREAD

**Discussion by andrewheiss** [andrewheiss/ath-quarto#27] [Blog comments]
Working with R, Cairo graphics, custom fonts, and ggplot | Andrew Heiss 
 http://www.andrewheiss.com/blog/2017/09/27/working-with-r-cairo-graphics-custom-fonts-and-ggplot/

**Comment by andrewheiss** [comment]
```header
Originally posted by Tod Robbins on 2017-10-04 16:18:31
````
A really nice use of Papyrus and Comic Sans!

**Reply by andrewheiss** [reply]
```header
Originally posted by Mara on 2017-10-21 16:44:10
````
Papyrus...
[https://youtu.be/jVhlJNJopOQ](https://youtu.be/jVhlJNJopOQ)

**Comment by andrewheiss** [comment]
```header
Originally posted by Carl Boettiger on 2017-11-23 04:10:48
````
Very nice post, remembering to use the Cario devices when using custom
fonts is a real gotcha. Speaking of alternative font choices for R
graphics, you may have seen
[https://github.com/hrbrmstr\...](https://github.com/hrbrmstr/hrbrthemes)
?

**Reply by andrewheiss** [reply]
```header
Originally posted on 2017-11-23 07:17:43
````
Yep! Those are great themes

**Comment by andrewheiss** [comment]
```header
Originally posted by Dipit Jaywant on 2019-06-30 07:20:19
````
I dont see any labels or digits in the plots for R ver 3.6.0 on R Studio
1.2.335 macbook air. Can you provide any solution ?

**Reply by andrewheiss** [reply]
```header
Originally posted on 2019-07-02 02:02:46
````
I\'m not sure---it works for me on 3.6.0 in macOS... Can you post a
reproducible example?

**Reply by andrewheiss** [reply]
```header
Originally posted by Dipit Jaywant on 2019-07-04 07:44:47
````
the problem is actually solved while removing R from my Mac i accidently
removed the Arial font family software that was pre installed ,so i just
had to install the font family software back, which I did and know i get
the plots with perfect labels.

Thanks for the prompt reply.

**Comment by andrewheiss** [comment]
```header
Originally posted by Hyebin Han 한혜빈 on 2021-12-22 17:21:23
````
OH MY GAWD THANK YOU SO MUCH FOR THIS\
I was experiencing the usual difficulties with downloading Cairo package
on my desktop R. I downloaded it via brew instead of install.packages
which was successful but it still did not connect with Rstudio. BUT this
method is so much easier AND IT WORKS!! Thanks fo much for this!!\
Hyebin\
Beginner-wannabe-bioinformatics-student

**Comment by jsavinc** [comment]
Note that for anyone reading this in 2023 and using ggplot version > 3.3.3, the `type` argument in `ggsave` has been deprecated and will now warn you that it's being ignored.

**Comment by andrewheiss** [comment]
With @jsavinc's comment above, it is still possible to make `ggsave()` use the Cairo PNG device with some slightly different syntax (which I'll update in the post soon). You can also have it use {ragg} PNGs instead, which also work great and use the correct DPI and dimensions:

```r
# With ragg
ggsave(p, filename = "example-cairo.png", 
       device = ragg::agg_png, res = 300,
       width = 4, height = 3, units = "in")

# With Cairo
ggsave(p, filename = "example-cairo.png", 
       device = png, type = "cairo", dpi = 300,
       width = 4, height = 3, units = "in")
```

**Reply by andrewheiss** [reply]
Updated! https://www.andrewheiss.com/blog/2017/09/27/working-with-r-cairo-graphics-custom-fonts-and-ggplot/

**Comment by yanxiang-wang** [comment]
Very helpful!!
